crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -191,7 +195,10 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                         empty_line().fmt(f)?;
                     }
                 }
-            } else if is_import_definition(preceding) && !is_import_definition(following) {
+            } else if (is_import_definition(preceding)
+                && !comments.has_trailing_own_line(preceding))
+                && (!is_import_definition(following) || comments.has_leading(following))
```

Comment:
Why is it necessary to check whether the next node has leading comments but is not for classes and functions?